site/app/entities/poll/Poll.php

```diff
@@ -91,39 +98,74 @@ public function getQuestion(): string {
     public function setQuestion(string $question): void {
         $this->question = $question;
     }
-
-    public function getStatus(): string {
-        return $this->status;
+    public function setClosed(): void {
+        $this->end_date = new \DateTime(DateUtils::BEGINING_OF_TIME);
     }
-
     public function setOpen(): void {
-        $this->status = "open";
+        $this->end_date = new \DateTime(DateUtils::MAX_TIME);
+    }
+    public function setEnded(): void {
+        $temp = DateUtils::getDateTimeNow();
+        $tempString = $temp->format('Y-m-d');
+        $this->end_date = new DateTime($tempString);
     }
-
     public function isOpen(): bool {
-        return $this->status === "open";
+        $now = DateUtils::getDateTimeNow();
+        return $now < $this->end_date;
     }
 
-    public function setClosed(): void {
-        $this->status = "closed";
+    public function isEnded(): bool {
+        $now = DateUtils::getDateTimeNow();
+        $closeDate = DateUtils::BEGINING_OF_TIME;
+        return $now > $this->end_date && $this->end_date->format('Y-m-d\TH:i:s') !== $closeDate;
     }
 
     public function isClosed(): bool {
-        return $this->status === "closed";
+        $now = DateUtils::getDateTimeNow();
+        return $now > $this->end_date && $this->end_date->format('Y-m-d\TH:i:s') === DateUtils::BEGINING_OF_TIME;
     }
 
-    public function setEnded(): void {
-        $this->status = "ended";
+    public function getDuration(): \DateInterval {
+        $seconds = $this->duration;
+        $years = floor($seconds / (365 * 24 * 60 * 60));
+        $seconds -= $years * (365 * 24 * 60 * 60);
+
+        $months = floor($seconds / (30 * 24 * 60 * 60));
+        $seconds -= $months * (30 * 24 * 60 * 60);
+
+        $days = floor($seconds / (24 * 60 * 60));
+        $seconds -= $days * (24 * 60 * 60);
+
+        $hours = floor($seconds / (60 * 60));
+        $seconds -= $hours * (60 * 60);
+
+        $minutes = floor($seconds / 60);
+        $seconds -= $minutes * 60;
+        return new \DateInterval("P{$years}Y{$months}M{$days}DT{$hours}H{$minutes}M{$seconds}S");
```

Comment:
getDuration() should only be expected to return the duration of the poll, in seconds, rather than wrapped as a `DateInterval` object - we should do this manipulation elsewhere if needed.

Reply:
The way I have it, outside the PHP Poll Class, the duration acts as a date interval. Due to database storing issues inside the class, I was forced to convert it to an integer that holds the seconds. The getters and setters are coded in a way such that the duration member variable appears as an int within the class but a date interval outside.